test/tests/functional/pbs_mom_hook_walltime.py

```diff
@@ -74,3 +74,62 @@ def test_mom_hook_not_counted_in_walltime(self):
                            offset=15)
         self.server.expect(JOB, {'resources_used.walltime': 5}, op=LE, id=jid,
                            extend='x')
+
+    def test_hold_time_not_counted_in_walltime(self):
```

Comment:
I don't know if I think this test is really required.  The hold state is a non-running state.  It's like saying, resources_used.walltime did not include time in the queued state.  The job hasn't started running, so it won't include it.  If you think it is required, then by all means keep it.

Reply:
I think this test should be added. This test validates that hold state as a non-running state will not affect the resources_used.walltime.

Reply:
I'll tell you what will make this test more interesting.  Submit the job.  Let it run for 10s.  Hold the job.  Rerun the job.  Let it sit held for 5s.  Release the hold and let it run to completion.  Of course now we're testing two things.  First that rerun wipes the resources_used.walltime, and two the whole held thing.  It might be better as two tests.